### PR TITLE
add support for comments on fragments and variables

### PR DESCRIPTION
**Pull request by vincent-duret**
@sungam3r I have fixed the issues for comment on fragments (spread and inline) and variables.
It should allow to uncomment the corresponding 3 tests in #1617

### CODE REVIEW

**Comment by pekkah** [comment]
This is related to #33 and to me looks wrong

**Comment by sungam3r** [comment]
#1617 ?

**Comment by vincent-duret** [comment]
@sungam3r https://github.com/graphql-dotnet/graphql-dotnet/pull/1617

File: src/GraphQLParser.Tests/ParserTests.cs [hunk 1]
```diff
@@ -12,6 +12,80 @@ public class ParserTests
...
+    }
+}";
+
+            var parser = new Parser(new Lexer());
+            var document = parser.Parse(new Source(query));
+            document.Definitions.Count().ShouldBe(1);
+            var def = document.Definitions.First() as GraphQLOperationDefinition;
+            def.VariableDefinitions.Count().ShouldBe(3);
+            def.VariableDefinitions.First().Comment.Text.ShouldBe("comment1");
+            def.VariableDefinitions.Skip(2).First().Comment.Text.ShouldBe("comment3");
```
Comment (sungam3r) [inline src/GraphQLParser.Tests/ParserTests.cs]:
```suggestion
            def.VariableDefinitions.First().Comment.Text.ShouldBe("comment1");
            def.VariableDefinitions.Skip(1).First().Comment.ShouldBeNull();
            def.VariableDefinitions.Skip(2).First().Comment.Text.ShouldBe("comment3");
```

File: src/GraphQLParser/ParserContext.cs [hunk 2]
```diff
@@ -87,20 +87,22 @@ private GraphQLFieldSelection CreateFieldSelection(int start, GraphQLName name,
             };
         }
 
-        private ASTNode CreateGraphQLFragmentSpread(int start)
+        private ASTNode CreateGraphQLFragmentSpread(int start, GraphQLComment? comment)
         {
             return new GraphQLFragmentSpread
             {
                 Name = ParseFragmentName(),
+                Comment = comment,
```
Comment (sungam3r) [inline src/GraphQLParser/ParserContext.cs]:
```suggestion
                Comment = comment,
                Name = ParseFragmentName(),
```

**Comment by sungam3r** [comment]
@pekkah Descriptions is another story. Nothing wrong with this PR. To support descriptions much more changes are needed. Someday we will do it.

**Comment by vincent-duret** [comment]
@sungam3r Thank you for your quick review and acceptance :-)
Do you have an estimate date of availbility on NuGet?

**Comment by sungam3r** [comment]
I have no rights to publish on NuGet. @joemcbride has. You can get the latest version from myget feed.

**Pull request merged**